libdnf/rpm/solv_query.cpp

```diff
@@ -194,98 +194,66 @@ Id what_downgrades(Pool *pool, Solvable * solvable) {
     return l;
 }
 
-}  //  namespace
-
-namespace libdnf::rpm {
-
-
-static inline bool name_compare_lower_id(const Solvable * first, Id id_name) {
+inline bool name_compare_lower_id(const Solvable * first, Id id_name) {
     return first->name < id_name;
 }
 
-static inline bool name_compare_icase_lower_id(const std::pair<Id, Solvable *> first, Id id_name) {
+inline bool name_compare_icase_lower_id(const std::pair<Id, Solvable *> first, Id id_name) {
     return first.first < id_name;
 }
 
-SolvQuery::SolvQuery(SolvSack * sack, InitFlags flags) : p_impl(new Impl(sack, flags)) {}
-
-SolvQuery::SolvQuery(const SolvQuery & src) : p_impl(new Impl(*src.p_impl)) {}
+}  //  namespace
 
-SolvQuery::~SolvQuery() = default;
 
-SolvQuery & SolvQuery::operator=(const SolvQuery & src) {
-    if (this == &src) {
-        return *this;
+SolvQuery::SolvQuery(SolvSack * sack, InitFlags flags) : PackageSet(sack) {
+    switch (flags) {
+        case InitFlags::EMPTY:
+            break;
+        case InitFlags::IGNORE_EXCLUDES:
+            // TODO(jmracek) add exclude application
+        case InitFlags::APPLY_EXCLUDES:
+        case InitFlags::IGNORE_MODULAR_EXCLUDES:
+        case InitFlags::IGNORE_REGULAR_EXCLUDES:
+            *pImpl |= sack->pImpl->get_solvables();
+            break;
     }
-    p_impl->query_result = src.p_impl->query_result;
-    p_impl->sack = src.p_impl->sack;
-    return *this;
-}
-
-SolvQuery & SolvQuery::operator=(SolvQuery && src) noexcept {
-    std::swap(p_impl, src.p_impl);
-    return *this;
 }
 
 SolvQuery & SolvQuery::operator|=(const SolvQuery & other) {
-    if (p_impl->sack != other.p_impl->sack) {
-        throw UsedDifferentSack("Cannot perform the action with SolvQuery instances initialized with different SolvSacks");
+    if (pImpl->sack != other.pImpl->sack) {
+        throw UsedDifferentSack(
+            "Cannot perform the action with SolvQuery instances initialized with different SolvSacks");
     }
-    p_impl->query_result |= other.p_impl->query_result;
+    *pImpl |= *other.pImpl;
     return *this;
 }
 
 SolvQuery & SolvQuery::operator&=(const SolvQuery & other) {
-    if (p_impl->sack != other.p_impl->sack) {
-        throw UsedDifferentSack("Cannot perform the action with SolvQuery instances initialized with different SolvSacks");
+    if (pImpl->sack != other.pImpl->sack) {
```

Comment:
Already discussed - Please could you move the check to PackageSet?

Reply:
I added commit which move the check to PackageSet.